Sources/FingerprintJSTests/Tests/IdentifierHarvesterTests.swift

```diff
@@ -146,4 +155,60 @@ final class IdentifierHarvesterTests: XCTestCase {
         let itemLabels = itemsTree.children?.map(\.label) ?? []
         XCTAssertTrue(itemLabels.isEmpty)
     }
+
+    func
+        test_givenBgIdentifierMissingLegacyIdentifierPresent_whenVendorIdentifier_thenMigratesOldIdentifierToNewStorage()
+    {
+        // given
+        let legacyIdentifier = UUID(uuidString: "a1084045-c9da-4e82-ae9b-8521379b0d07")!
+        identifierStorableSpy.loadIdentifierReturnDictionary = [
+            "vendorIdentifier": legacyIdentifier
+        ]
+
+        // when
+        let identifier = sut.vendorIdentifier
+
+        // then
+        XCTAssertEqual(
+            identifierStorableSpy.storeIdentifierInputs,
+            ["vendorIdentifierBackground": legacyIdentifier]
+        )
+        XCTAssertEqual(identifier, legacyIdentifier)
+    }
+
+    func test_givenBgIdentifierPresent_whenVendorIdentifier_thenReturnsBgIdentifier() {
+        // given
+        let bgIdentifier = UUID(uuidString: "a1084045-c9da-4e82-ae9b-8521379b0d07")!
+        identifierStorableSpy.loadIdentifierReturnDictionary = ["vendorIdentifierBackground": bgIdentifier]
+
+        // when
+        let identifier = sut.vendorIdentifier
+
+        // then
+        XCTAssertEqual(identifier, bgIdentifier)
+    }
+
+    func test_givenNoStoredIdentifiers_whenVendorIdentifier_thenStoresAndReturnsSystemVendorIdentifier() {
+        // given
+        let systemIdentifier = UUID(uuidString: "a1084045-c9da-4e82-ae9b-8521379b0d07")!
+        vendorIdentifierProvidableSpy.vendorIdentifierReturnValue = systemIdentifier
+
+        // when
+        let identifier = sut.vendorIdentifier
+
+        // then
+        XCTAssertEqual(identifier, systemIdentifier)
+        XCTAssertEqual(
+            identifierStorableSpy.storeIdentifierInputs,
+            ["vendorIdentifierBackground": systemIdentifier]
+        )
+    }
+
+    func test_givenNoIdentifiers_whenVendorIdentifier_thenReturnsNil() {
+        // given, when
+        let identifier = sut.vendorIdentifier
+
```

Comment:
Or we could make it explicit, just to be absolutely clear.
```suggestion
        // given
        identifierStorageSpy.loadIdentifierReturnDictionary = [:]
        vendorIdentifierProviderSpy.vendorIdentifierReturnValue = nil
        
        // when
        let identifier = sut.vendorIdentifier

```